tests/queries/0_stateless/02494_query_cache_udf_executable.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+# Tags: no-parallel
+# Tag no-parallel: Messes with internal cache
+
+# Test for issue #77553: External UDFs may be non-deterministic. The query cache should treat them as such, i.e. reject them.
+# Also see 02494_query_cache_udf_sql.sql
+
+set -e
```

Comment:
As per your findings in https://github.com/ClickHouse/ClickHouse/issues/59988#issuecomment-2727041022 (thanks for noticing!), we'll need to go for an integration test actually. I tried quickly with standard `clickhouse-client` instead of `clickhouse-local` but could not get it to work easily.

Re integration tests: They can be a bit more difficult to setup, see tests/integration/README.md for prerequisites. I recommend testing with the runner script.